### ISSUE TITLE
FR: Setting of device ID at runtime

### ISSUE THREAD

**Issue by jaredwolff** [golioth/golioth-zephyr-sdk#143]
### Hardware Target

nRF9160

### Golioth Dsdk Commit Hash

main

### What Host OS are you using?

Mac

### What happened?

Currently TLS_PSK_ID is set at compile time. In some cases, someone may want to use an onboard unique I identifier (like an IMEI) as the device ID. Setting this during Golioth init would be handy for this purpose (since that IMEI is not retrievable until runtime) 

Happy to provide more context but hope that helps.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by vitprajzler** [comment]
Thanks for the feedback @jaredwolff!

I'm definitely in favor of adding the option to get the Device ID / PSK ID at runtime to the SDK.

In general, I would like to enable a workflow that's easy for users to follow. The plan is to have two different workflows, one for development, and one for production. The "runtime" TLS_PSK_ID to me sounds more like the production workflow. 

Could you share what a good workflow would be for you? 
Do you see the workflow changing between development and production at scale?

**Background:** 
Currently, as a Golioth user, you need to synchronize the TLS_PSK_ID and the TLS_PSK between the Zephyr device and the Golioth cloud. With our last update, the TLS_PSK_ID now includes both the Device ID and the Project ID.

So although extracting the TLS_PKS_ID at runtime would help the runtime, you (as a user) still need to

- extract the ID from the device and register it in the cloud
- get project ID from the cloud and configure it on the device
- synchronize the TLS_PSK between the device and the cloud

**Comment by jaredwolff** [comment]
> Could you share what a good workflow would be for you?

This is my current production and dev workflow on my own setup. I'm not sure why it would be different in either case. 

> Do you see the workflow changing between development and production at scale?

Nope.

> With our last update, the TLS_PSK_ID now includes both the Device ID and the Project ID.

Ahh ok. That's fine and actually is preferred. Only one thing to set instead of two.

**Comment by jaredwolff** [comment]
With the above being said, the PSK_ID doesn't need to be set in the latest?

**Comment by vitprajzler** [comment]
Sorry, forgot to answer your question. Both `PSK` and `PSK_ID` need to be set. The `PSK_ID` just got more structure to it (includes both project ID and the device ID).

**Comment by sam-golioth** [comment]
Hey @jaredwolff! I wanted to follow up on this. With the release of v0.7.0 of the Zephyr SDK, we've moved hardcoded PSKs out of the SDK and in general are much less opinionated on how credentials (PSK or certificates) are set. The SDK now expects that the user will have loaded credentials into the [Zephyr TLS credential store](https://docs.zephyrproject.org/apidoc/latest/group__tls__credentials.html) before it attempts to connect. We've moved the hardcoded credentials and the runtime settings module into the samples, so that's still available (with renamed Kconfig options) if you'd like to use that, but you're also now free to use whatever method you'd like to store and retrieve credentials, so long as they eventually end up in the TLS credential store.
